### ISSUE TITLE
Add HTTP/1.1 Response Builder

### ISSUE THREAD

**Issue by jemc** [jemc/mare#82]
We have basic HTTP/1.1 request parsing in our fledgling HTTP server package, so now it's time to move on to response building.

That is, we want to add a `ResponseBuilder` class to `packages/http/server`.

**Comment by jemc** [comment]
Here's a proposed implementation plan to walk you through how I would approach it. This is a simple approach to start, and after it's done I think we'll both have ideas on how to enhance it by fiddling with the interface and/or implementation: 

---

- Create a `:class ResponseBuilder` in `packages/http/server` (in the future we may make this a `:struct` instead for better performance, but I'm still ironing kinks with `:struct` so I say avoid getting snagged on that for now)

- Write a constructor for it that takes a `target ByteStreamWritable` (a trait/interface that has the `<<` operator for appending chunks of `Bytes` to a logical stream. It can store this parameter in a field for persisting it across calls.

- Add `:fun ref` methods for doing things like setting status codes, adding a header, or writing to the response body. Each of these methods should append `Bytes` chunks to the aforementioned `target`. 

- Consider using named methods for each known HTTP status code (e.g. `status_not_found`, `status_forbidden`, `status_ok`, etc.), since this is convenient for the caller and will let the compiler prevent mistakes with unknown status codes by complaining about a missing method for that name.

- Write tests. In the tests you should find it convenient to use a `Bytes.new` as the `target` passed into the constructor, since that `ByteStreamWritable` is compatible with a plain `Bytes` because `Bytes` implements the `<<` method. So you should be able to use `@assert = bytes == b"[whatever content is expected]"` at the end of each test to check what the written response was.

- You may notice that with this approach, if you call the methods in the wrong order (like writing headers prior to writing a status code line), then you end up with a junk response content. In the future we will try out some type system tricks to prevent this, but for now I suggest leaving it simple and we can come back to this to improve it in a fancier way in the future.

---

- Once this is in place, you may also want to update the HTTP server example in `examples/http`. Replace the simple bytes string literal response content with a `ResponseBuilder` that uses the `io.write_stream` as its `target` in its constructor.

- Confirm that you can run the example it with a command like this one, which is similar to what I use for it:
```
make example extra_args="--backtrace --print-perf" dir="examples/http"
```

- Confirm that while the example is running, it responds to HTTP requests on localhost port 8080. I use `curl` or sometimes the `wrk` command to hammer it with a lot of requests (although currently the performance won't be ideal because we're compiling in debug mode rather than release mode, so we're still missing LLVM optimizations - we will get this working in the future). Here's an example `wrk` command just as a proof of concept rather than as a serious test:
```
wrk -H 'Connection: keep-alive' --connections 256 --threads 1 --duration 5 --timeout 1s http://localhost:8080/example
```

**Comment by tonchis** [comment]
Closing now that we merged #83
